compiler/passes/normalize.cpp

```diff
@@ -4870,6 +4873,47 @@ static void addToWhereClause(FnSymbol*  fn,
   combine->insertAtTail(test);
 }
 
+static void fixupExplicitGenericVariables(CallExpr* call) {
+  // fixup the pattern `(CallExpr _domain ?)` to be `(_domain(?))`,
+  // marking the `DefExpr` as generic
+
+  // if its a call like `_domain ?`, make it `_domain(?)` and MARKED_GENERIC
+  SymExpr* symExpr = nullptr;
+  bool actIsQuestion = false;
+  if (auto se = toSymExpr(call->baseExpr)) {
+    // for some reason this breaks with ranges, skip since they aren't a problem currently
```

Comment:
It's because ranges are a type with defaults. You need something like

```
  bool genericWithDefaults = false;
  if (AggregateType* at = toAggregateType(type))
    genericWithDefaults = at->isGenericWithDefaults();
```

and then to not do the change for generic-with-defaults.

Reply:
That makes sense. And making this change probably means I no longer need to update `test/types/records/generic/generic-with-defaults-q-problem.bad`.

Reply:
Also I forgot to say, it's important to label whatever you do here as a workaround. One issue is that at this point in compilation, only incomplete type information is known.